android/autodispose-android-archcomponents/src/main/java/com/uber/autodispose/android/lifecycle/KotlinExtensions.kt

```diff
@@ -77,3 +90,93 @@ inline fun Lifecycle.scope(
   boundaryResolver: CorrespondingEventsFunction<Event>
 ): ScopeProvider = AndroidLifecycleScopeProvider.from(
     this, boundaryResolver)
+
+/**
+ * Extension that proxies to [Flowable.as] + [AutoDispose.autoDisposable]
+ */
+@CheckReturnValue
+inline fun <T> Flowable<T>.autoDisposable(lifecycleOwner: LifecycleOwner): FlowableSubscribeProxy<T> =
+    this.`as`(AutoDispose.autoDisposable(AndroidLifecycleScopeProvider.from(lifecycleOwner)))
+
+/**
+ * Extension that proxies to [Flowable.as] + [AutoDispose.autoDisposable] and takes an [untilEvent] when
+ * subscription will be disposed.
+ */
+@CheckReturnValue
+inline fun <T> Flowable<T>.autoDisposable(lifecycleOwner: LifecycleOwner, untilEvent: Event): FlowableSubscribeProxy<T> =
```

Comment:
Let's default the untilEvent to something sensible and then just have one overload?

Reply:
Can we really default an `untilEvent` reliably for everyone? Usually the `untilEvent` is inferred from when in the lifecycle you subscribe. However that inference logic is private in `AndroidLifecycleScopeProvider`.

Reply:
I'd be fine with it package private

Reply:
ec8cd98